Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Well I meant that too every force(real life force(air friction,Magnus effect or anything that can be related to flying things)) should be handled in separate function so you can call it in every process and do the right logic if it is actual or not.